src/daemon/mod.rs

```diff
@@ -695,14 +694,170 @@ fn create_password(prompt: &str) -> std::io::Result<String> {
+                }
+            }
+            Err(err) => warn!("Error loading car DB at {}: {err}", file.display()),
+        };
+    }
+
+    // TODO: use `--consume-snapshot` CLI option once it's implemented
+    let mut consume_snapshot_file = false;
+    if config.client.snapshot_path.is_none() {
+        let epoch = {
+            if store.read_only_len() == 0 {
+                0
+            } else if let Ok(Some(ts)) = Tipset::load_heaviest(&store, store.writer().as_ref()) {
+                ts.epoch()
+            } else {
+                0
+            }
+        };
+        fetch_snapshot_if_required(config, epoch, opts.auto_download_snapshot, &db_root_dir)
+            .await?;
+        consume_snapshot_file = true;
+    }
+
+    if !opts.skip_load.unwrap_or_default() {
+        if let Some(path) = &config.client.snapshot_path {
+            let (car_db_path, ts) =
+                import_chain_as_forest_car(path, &forest_car_db_dir, consume_snapshot_file).await?;
+            heaviest_tipset = Some(ts);
+            store.read_only_files(std::iter::once(car_db_path.clone()))?;
+            info!("Loaded car DB at {}", car_db_path.display());
+        }
+    }
+
+    Ok((Arc::new(store), heaviest_tipset))
+}
+
+async fn import_chain_as_forest_car(
+    from_path: &Path,
+    forest_car_db_dir: &Path,
+    consume_snapshot_file: bool,
+) -> anyhow::Result<(PathBuf, Tipset)> {
+    info!("Importing chain from snapshot at: {}", from_path.display());
+
+    let stopwatch = time::Instant::now();
+
+    let downloaded_car_temp_path =
+        tempfile::NamedTempFile::new_in(forest_car_db_dir)?.into_temp_path();
+    let temp_file_ready = if from_path.is_file() && consume_snapshot_file {
+        if let Err(err) = fs::rename(from_path, &downloaded_car_temp_path) {
+            warn!(
+                "Failed to rename file from {} to {}: {err}",
+                from_path.display(),
+                downloaded_car_temp_path.display()
+            );
+            false
+        } else {
+            true
+        }
+    } else {
+        false
+    };
+
+    if !temp_file_ready {
+        if from_path.is_file() {
+            std::fs::copy(from_path, &downloaded_car_temp_path)?;
+        } else {
+            let url = Url::parse(&from_path.display().to_string())?;
+            snapshot::download_file(
+                url,
+                forest_car_db_dir,
+                downloaded_car_temp_path
+                    .file_name()
+                    .and_then(OsStr::to_str)
+                    .context("Infallible getting file name")?,
+            )
+            .await?;
+        }
+    }
+
+    let (is_forest_car, ts) = {
+        let car = AnyCar::new(RandomAccessFile::open(&downloaded_car_temp_path)?)?;
+        let ts = car.heaviest_tipset()?;
+        (matches!(car, AnyCar::Forest(_)), ts)
+    };
+
+    let forest_car_db_path =
+        forest_car_db_dir.join(format!("{}{FOREST_CAR_FILE_EXTENSION}", ts.epoch()));
```

Comment:
This could clash with other files.

Reply:
If it's the same snapshot, I think it should not matter. If it's a diff car file, I wanted to name it `{from_epoch}-{to_epcoh}.forest.car.zst`, and I'd like to discuss about injecting the epoch range info to the diff car header. Basically 

```rust
let cid = store.put((TipsetKeys, TipsetKeys, GenesisTipsetKeys))
output_car.roots = vec![cid]
```

Reply:
Changed filename to {unix_timestap}.forest.car.zst